### PR TITLE
fix: Add path traversal protection to IPC file handlers

### PR DESCRIPTION
**Pull request by e-fisher**
## Description

IPC handlers that read or write files accept paths from the renderer without validating they stay within their expected workspace directory. A compromised renderer could send `../../../etc/passwd` or an absolute path outside the workspace to read arbitrary files, execute arbitrary scripts, or write to unexpected locations.

This adds `ensureWithinDirectory` to `src/utils/path.ts`, which resolves the path and throws if it escapes the base directory. It handles case-insensitive comparison on macOS/Windows and prevents sibling-prefix attacks (`/foo/barbaz` vs `/foo/bar`).

Applied to 8 call sites across 6 handler files: dataFiles, HAR, generator (Open + Save), browserTest (Open + Save), script (Run + RunFromGenerator), and cloud. `ScriptHandler.Open` and `ScriptHandler.Save` are intentionally excluded because they support external scripts selected via native file dialog.

## How to Test

1. Open a recording, generator, browser test, and script from the file list
2. Generate and validate a script
3. Save a generator and browser test
4. Run a script in the cloud

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests for my changes.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)

Part of security hardening for critical findings from a codebase audit.

---

> [!NOTE]
> **Medium Risk**
> Reduces path-traversal risk on main-process file I/O but leaves script run/open/save paths less restricted; behavior changes only when malicious or malformed paths are sent.
> 
> **Overview**
> Adds **`ensureWithinDirectory`** in `src/utils/path.ts` so renderer-supplied paths are resolved under a workspace root and rejected when they escape (e.g. `../`, absolute paths outside the base, or sibling-prefix tricks), using the existing **`key()`** normalization for case-insensitive platforms.
> 
> **Open/Save/Load** IPC handlers for **browser tests**, **generators**, **HAR recordings**, and **data file previews** now resolve paths through this helper instead of trusting absolute paths or naive `join`. **Unit tests** cover traversal, out-of-tree absolutes, and prefix edge cases.
> 
> **`ScriptHandler.Run`** only drops a redundant absolute-path branch; it does **not** use `ensureWithinDirectory` in this diff (and **Open/Save** remain unchanged for external scripts).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 21bc11b33dd74c906f679e9f58a6c8648d0913fb. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/handlers/script/index.ts [hunk 1]
```diff
@@ -115,13 +115,17 @@ export function initialize() {
       console.info(`${ScriptHandler.RunFromGenerator} event received`)
 
       const browserWindow = browserWindowFromEvent(event)
+      const resolvedScriptPath = path.ensureWithinDirectory(
+        SCRIPTS_PATH,
+        scriptPath
+      )
```
[the remark this answers is not in the document]
Comment (e-fisher) [reply src/handlers/script/index.ts]: Real bug. RunFromGenerator uses temp paths from app.getPath('temp'), not SCRIPTS_PATH. Removed ensureWithinDirectory from RunFromGenerator — temp paths come from the main process IPC handler.

File: src/handlers/script/index.ts [hunk 2]
```diff
@@ -65,10 +65,10 @@ export function initialize() {
...
         await waitForProxy()
 
-        const absolute = path.isAbsolute(scriptPath)
-        const resolvedScriptPath = absolute
-          ? scriptPath
-          : path.join(SCRIPTS_PATH, scriptPath)
+        const resolvedScriptPath = path.ensureWithinDirectory(
+          SCRIPTS_PATH,
+          scriptPath
+        )
```
[the remark this answers is not in the document]
Comment (e-fisher) [reply src/handlers/script/index.ts]: Real bug. Reverted ScriptHandler.Run and CloudHandlers.Run to the original path resolution pattern since external scripts selected via native file dialog have paths outside SCRIPTS_PATH.

**Comment by e-fisher** [comment]
bugbot run

File: src/utils/path.ts [hunk 3]
```diff
@@ -46,6 +46,21 @@ export function equal(a: string, b: string): boolean {
...
+  const resolved = resolve(baseDir, filePath)
+  const normalizedBase = resolve(baseDir)
+  if (
+    key(resolved) !== key(normalizedBase) &&
+    !key(resolved).startsWith(key(normalizedBase + sep))
+  ) {
+    throw new Error('Path is outside allowed directory')
+  }
+  return resolved
+}
```
[the remark this answers is not in the document]
Comment (e-fisher) [reply src/utils/path.ts:62]: Real bug. pathe's resolve always returns forward-slash paths, but sep is backslash on Windows. Changed to hardcoded '/' since the project uses posix paths internally (see #1248).

**Comment by e-fisher** [comment]
bugbot run

**Review by allansson — COMMENTED** [review]
Coming from the perspective of someone who is just about to push a PR for opening and saving tests from any location 😬:

Opening a test (generator or browser) and running it is the same as running a script. It's an act of trust. We can't limit the directories from where a file can be opened without severely limiting the freedom of users to organize files in the way that fits them. Since the user is picking the file it is a clear sign that the user trusts it (even if they might be careless.)

The problems are the embedded paths in generators. These are not immediately known to the user. 

Referencing recordings is benign, I think. They contain hardcoded data that will become hardcoded requests (it's the rules that bring runtime dynamics.)

Data files however is where you can load data dynamically, e.g. `open("/etc/passwd")`, and send it somewhere. This is what we should protect against.

Relative paths are a problem right now because an attacker can assume where the generator is stored e.g. `~/Documents/k6-studio/Generators`. If the generator can be stored in an arbitrary location they can only:

* use them and hope for the best, or
* use absolute paths

So here's what I think we should enforce:

1. Always warn about absolute paths. We know that k6 Studio doesn't store them as absolute at rest, so seeing one means that the file has been tampered with.
2. If a file is opened **_inside_** a workspace and a file reference goes outside of the workspace root, relative or absolute, then the user should be warned about opening the file.
3. If a file is opened **_outside_** of the workspace, then we don't have a point of reference for relative paths so we'll have to either:
    1. trust them completly,
    2. treat the path of the file as the workspace root, effectively forbidding `../` traversal above the file, or
    3. always warn the user about the files which files are being referenced
5. If we get to supporting workspaces, do what vscode does: ask if the user trusts it before opening it.
6. Keep a list of blacklisted directories, e.g. `<DRIVE>:\Windows`, everything except the `/usr/` and `/User`, dot-folders like `~/.ssh`, etc.

The warning implementation could go like this:

```ts
// handler

if (!confirmed && hasSuspiciousPaths(generator)) {
  return {
    status: "warn-paths",
    paths: [
      "/etc/passwd"
    ]
  }
}

return {
  status: "opened",
  content: generator
}

// ui

const handleOpenFile = (filePath) => {
  void window.studio.fs.openFile(filePath)
    .then(result => {
      if (result.status === "warn-paths") {
        setShowWarnPaths(result.paths)
      }

      navigate(...)
    })
}

const handleConfirmOpen = (filePath) => {
  // Explicitly tell openFile that it's ok.
  void window.studio.fs.openFile(filePath, { confirm: true })
}
```

Lastly, how can a renderer be compromised in a way that the main script can't? If the main script is compromised they can just remove these checks anyway. We can't ever protect users from downloading compromised builds of the app. If we are worried about XSS attacks, then we should work to find and fix those places instead.

**Comment by e-fisher** [comment]
Let's shelve this for now and revisit when we have workspace support
